### ISSUE TITLE
bug(narrative-llm): ICD chronic-condition code↔label mismatch in LLM output (G20 → "glaucoma")

### ISSUE THREAD

**Issue by TomoOkuyama** [TomoOkuyama/clinosim#1445]
## Summary

The LLM narrative pipeline (vLLM + Qwen3.8-27B-FP8) produces `<label> (<CODE>)` phrasing that mixes an ICD code with an incorrect display in the US locale — most notably rendering `G20` (Parkinson's disease) as **"glaucoma"** (a completely different condition, correctly coded H40). The label side is hallucinated: the CIF carries the code with no display attached and the model resolves the display from parametric knowledge, sometimes wrong. Observed on a p=100 s=400 H100 run.

## Repro

```
clinosim simulate --country US -p 100 -s 400 --start 2025-09-15 --end 2026-09-15 --format fhir-r4 -o /tmp/us-100
clinosim narrate --cif-dir /tmp/us-100/cif --provider vllm --llm-config /path/to/vllm.yaml --country US --set-current --concurrency 32
```

Then walk `cif/narratives/vllm/documents/*/*.json`, grep the assembled section text, and match `<label> (<CODE>)` pairs against ICD-10-CM ground truth.

## Sample hallucination (highest severity)

CIF for `POP-000019` carries:

```
chronic_conditions:
  - code: G20    system: icd-10-cm   onset: 2018-04-08
  - code: N18    ...                                (chronic kidney disease)
  - code: I10    ...                                (essential hypertension)
  - code: C34    ...                                (lung cancer)
  - code: E66.3  ...                                (obesity)
current_medications:
  - Levodopa/Carbidopa 100/25mg PO tid              (antiparkinson)
```

The LLM narrative renders the same patient as:

> "The patient is an 89-year-old female with a history of essential hypertension (I10), chronic kidney disease (N18, CKD G4), **glaucoma (G20)**, lung cancer (C34), and obesity (E66.3)."

Clinical impact: a downstream reader sees "glaucoma" as the reason the patient takes Levodopa/Carbidopa — a nonsensical pairing. The narrative silently disconnects the medication from its indication (Parkinson's), which is precisely the kind of consumer-visible clinical incoherence the sim is supposed to avoid.

## Full scan on p=100 US (519 LLM narrative docs)

Substring-audit matching `<label> (<CODE>)` where the code is documented in ICD-10-CM ground truth (partial list):

| Code | Truth | Model output | Count |
|---|---|---|---|
| G20 | Parkinson's disease | **glaucoma** | 3 |
| E11 | Type 2 diabetes mellitus | "diabetes mellitus" / "diabetes" | 51 |
| M17 | Osteoarthritis of knee | "osteoarthritis" | 22 |
| F32 | Major depressive disorder, single episode | "depression" / "mild depression" | 31 |
| J44 | Chronic obstructive pulmonary disease | "copd" | 6 |
| F33 | Recurrent depressive disorder | "depression" | 2 |
| Z38 | Single liveborn | "born infants according to place of birth" | 1 |

Two shapes to distinguish:

- **Hallucinated label (G20 → glaucoma)** — the model asserts a display that is factually wrong for the code. Consumer is misled. 3/519 docs (~0.6 %) on this seed.
- **Over-simplified label (E11 → "diabetes", M17 → "osteoarthritis")** — the model drops the code's specificity (type / anatomic site / episode). Not a hallucination in the strict sense; the reader who ignores the code and reads only the label loses information but is not misled toward a different disease. ~110/519 docs (~21 %) on this seed.

## JP scan

Same repro on JP p=100 s=400 (819 docs) produced **0 mismatches** on the same audit. The LLM's JP training data appears to keep the ICD-code↔label pairing accurate; the miss is US-only under the current prompt shape.

## Root cause (hypothesis)

The chronic_conditions payload sent to the LLM contains the code only:

```
chronic_conditions:
  - {code: G20, system: icd-10-cm, ...}
```

with **no display field**. The LLM resolves `G20 → "glaucoma"` from parametric memory, occasionally incorrectly. `clinosim.codes.lookup("icd-10", "G20", "en")` in the codebase already returns `"Parkinson's disease"`, so the resolution *exists* — it just is not injected into the prompt.

The bundled prompt's Rule 2 says "cite codes verbatim" but does not require the model to use a specific display, which leaves the label resolution up to the model.

## Fix design (proposal — not implemented yet)

Two options:

1. **Prompt-side fix**: extend the chronic_conditions serialization in the narrative context builder to include a `display_en` / `display_ja` resolved via `code_lookup`, and add a Rule N item in `narrative_seed_bundle.yaml` — "When citing an ICD code, use the paired `display` field verbatim; never resolve the label from your own knowledge."
2. **Post-process check**: after narrate, walk sections, extract `<label> (<CODE>)` pairs, and reject-and-retry when the label does not match `code_lookup(system, code, lang)` within a fuzzy-match threshold. Heavier; useful as a defense-in-depth layer.

Recommendation: start with option 1 (prompt-side). It is the smallest change and directly addresses the missing input rather than adding a validator. Option 2 belongs behind option 1 as a safety net.

## Not-in-scope

- Statistical drift in per-seed counts.
- The label over-simplification cases (E11 → "diabetes") — those are a separate readability concern and belong in a different issue if the maintainer wants to prescribe verbatim display use for those codes too.
